### PR TITLE
[android] include ds2 as part of the Android SDK

### PR DESCRIPTION
**Pull request by andrurogerz**
Builds all 4 Android variants of compnerd/ds2 to be included in the Android SDK.

NOTE: ds2 is not yet referenced by `swift/swift-installer-scripts`, so this change does not have any effect other than validating the build.

### Test Plan
Manually inspected artifacts produced by the Android ds2 steps:
```
C:\Users\Andrew\Downloads>unzip -l ds2-Android-arm64.zip
Archive:  ds2-Android-arm64.zip
  Length     EAs   ACLs    Date   Time    Name
 --------    ---   ----    ----   ----    ----
 11907472      0      0  08/24/24 17:31   bin/ds2
 --------  -----  -----                   -------
 11907472      0      0                   1 file

C:\Users\Andrew\Downloads>unzip -l ds2-Android-armv7.zip
Archive:  ds2-Android-armv7.zip
  Length     EAs   ACLs    Date   Time    Name
 --------    ---   ----    ----   ----    ----
 10356780      0      0  08/26/24 15:19   bin/ds2
 --------  -----  -----                   -------
 10356780      0      0                   1 file

C:\Users\Andrew\Downloads>unzip -l ds2-Android-i686.zip
Archive:  ds2-Android-i686.zip
  Length     EAs   ACLs    Date   Time    Name
 --------    ---   ----    ----   ----    ----
 10048944      0      0  08/26/24 15:19   bin/ds2
 --------  -----  -----                   -------
 10048944      0      0                   1 file

C:\Users\Andrew\Downloads>unzip -l ds2-Android-x86_64.zip
Archive:  ds2-Android-x86_64.zip
  Length     EAs   ACLs    Date   Time    Name
 --------    ---   ----    ----   ----    ----
 11268640      0      0  08/26/24 15:21   bin/ds2
 --------  -----  -----                   -------
 11268640      0      0                   1 file
```

### CODE REVIEW

File: .github/workflows/swift-toolchain.yml [hunk 1]
```diff
@@ -298,6 +302,155 @@ jobs:
...
+
+      - uses: actions/download-artifact@v4
+        with:
+          name: windows-regsgen2
+          path: ${{ github.workspace }}/BinaryCache/RegsGen2
+
+      - uses: nttld/setup-ndk@v1
+        id: setup-ndk
+        with:
+          ndk-version: r26b
```
Comment (Steelskin) [inline .github/workflows/swift-toolchain.yml:422]: Could you move this to an input so we only set it once? It's used in a few places.

**Review by Steelskin — APPROVED** [review]

**Review by Steelskin — APPROVED** [review]
Hold on, I just noticed the repo. This should probably be a compnerd PR.

**Comment by andrurogerz** [comment]
> Hold on, I just noticed the repo. This should probably be a compnerd PR.

@Steelskin Thanks for taking a look! Do you mean it should be reviewed by compnerd or should go to `compnerd/swift-build`?

**Comment by Steelskin** [comment]
> > Hold on, I just noticed the repo. This should probably be a compnerd PR.
> 
> @Steelskin Thanks for taking a look! Do you mean it should be reviewed by compnerd or should go to `compnerd/swift-build`?

It should go in the `compnerd/swift-build` repo. We'll merge it after.

**Comment by andrurogerz** [comment]
> It should go in the `compnerd/swift-build` repo. We'll merge it after.

@Steelskin I forked `compnerd/swift-build` and these changes cherry-picked cleanly. New PR with the same content is at https://github.com/compnerd/swift-build/pull/820. Thanks!

**Pull request closed without merging**